stablehlo/transforms/StablehloComplexMathExpander.cpp

```diff
@@ -0,0 +1,76 @@
+/* Copyright 2024 The StableHLO Authors. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+#include "stablehlo/dialect/StablehloOps.h"
+#include "stablehlo/transforms/PassUtils.h"
+#include "stablehlo/transforms/Passes.h"
+
+namespace mlir {
+namespace stablehlo {
+#define GEN_PASS_DEF_STABLEHLOCOMPLEXMATHEXPANDERPASS
+#include "stablehlo/transforms/Passes.h.inc"
+
+namespace {
+
+static Value getConstantLikeMaxFiniteValue(OpBuilder &b, Location loc,
+                                           Value val) {
+  auto ty = cast<FloatType>(getElementTypeOrSelf(val.getType()));
+  return getConstantLike(
+      b, loc, llvm::APFloat::getLargest(ty.getFloatSemantics()), val);
+}
+
+//===----------------------------------------------------------------------===//
+// Pass
+//===----------------------------------------------------------------------===//
+
+struct StablehloComplexMathExpanderPass
+    : public impl::StablehloComplexMathExpanderPassBase<
+          StablehloComplexMathExpanderPass> {
+  StablehloComplexMathExpanderPass()
+      : StablehloComplexMathExpanderPassBase<
+            StablehloComplexMathExpanderPass>() {}
+
+ public:
+  LogicalResult initialize(MLIRContext *context) override {
+    config.useTopDownTraversal = true;
+    RewritePatternSet patterns_(context);
+    populateStablehloComplexMathExpanderPatterns(&patterns_, context);
+    patterns = std::move(patterns_);
+    return success();
+  }
+
+  void runOnOperation() override {
+    auto func = getOperation();
+    if (failed(applyPatternsAndFoldGreedily(func, patterns, config))) {
+      func.emitError("Failed to converge StableHLOComplexMathExpanderPass in ")
+          << config.maxIterations << " iterations";
+      signalPassFailure();
+    }
+  }
+
+ private:
+  FrozenRewritePatternSet patterns;
+  GreedyRewriteConfig config;
+};
+
+#include "stablehlo/transforms/StablehloComplexMathExpanderPatterns.h.inc"
+
+}  // namespace
+
+void populateStablehloComplexMathExpanderPatterns(RewritePatternSet *patterns,
+                                                  MLIRContext *context) {
+  patterns->add<Log1pOp_ComplexElementType_ComplexMathExpander>(context);
```

Comment:
Lets instead use the `populateWithGenerated` which should be included by `#include "stablehlo/transforms/StablehloComplexMathExpanderPatterns.h.inc"`

Similar to https://github.com/openxla/stablehlo/blob/9e0c9e34d02af255bfe8f896ba7dc910758c6ecf/stablehlo/transforms/VhloToVersion.cpp#L405

This way new patterns in the TD file will be auto added to the pass

Reply:
Nice! Applied.